application_components_201.rst

```diff
@@ -1,11 +1,114 @@
+Messages are send to different type of exchanges.
+While Direct, Fanout, Topic and Header Exchanges are part of the core system, others plugins are rooted in the community. 
+No prior understanding of the erlang programming language is needed to setup, configure and operate the message broker.
+The documentation on the projects homepage reads a fair amount of different configurations with explanations.
+
+To setup a cluster a minimum of two nodes is required.
+All nodes of a cluster must be in the same ip network segment.
+It is possible to send huge messages via AMQP to a RabbitMQ node.
+The maximum message size is limited by the amount of available RAM on the node.
+Depending on the durability of a message and the policy of queues, the messages are synchronized to other nodes.
+A disc-node ensures a message is stored to a disc before delivering.
+Memory based nodes provide a much higher message throughput.
+To establish an encrypted communication between nodes, IPSec or TLS can be used.
+Authentication is possible via an internal database, LDAP, SASL and PKI client certificates.
+With a PKI in place consumers and publisher can share the same authorization.
+No preshared password among all processes is needed, but different private keys are mandatory.
+This is accomplished by using the commonname of the dn in the x509 client certificate as  username.
+Different applications can be separated by the concept of virtual hosts in a similar way the Apache web server does.
+Configuration is provided by webGUI or commandline tools.
+Programming libraries and tools for a wide range of environments are available.
+
+The installation from the Debian repository leads to a configuration with a single host.
+Similar installation instructions are provided for Windows, Solaris and other operation systems.
+
+.. code-block:: bash
+
+  curl 'https://www.rabbitmq.com/rabbitmq-signing-key-public.asc' | sudo apt-key add -
+  echo 'deb http://www.rabbitmq.com/debian/ testing main' | sudo tee /etc/sources.d/rabbitmq.list
+  sudo apt-get update
+  sudo apt-get upgrade
+  sudo apt-get install rabbitmq-server
+
+To backup and restore configuration data the management plugin should be configured.
+
+.. code-block:: none
+
+  { rabbitmq_management, [
+      {  listener, [ { port, 15672 },
+                     { ssl, true },
+                     { ssl_opts, [ { cacertfile, "/etc/ssl/certs/cacert.crt" },
+                                   { certfile,   "/etc/ssl/certs/node1.crt" },
+                                   { keyfile,    "/etc/ssl/private/node1.key" } ]}
+                   ]} // configured listener
+  ]} // configured rabbitmq_management
+  ]. // EOC
+
+.. code-block:: console
+
+  rabbitmqadmin export rabbit.config
+  rabbitmqadmin -q import rabbit.config
+
+Show a detailed report about queues, users and connections
+
+.. code-block:: console
+
+  rabbitmqctrl report
```

Comment:
I think this should be `rabbitmqctl`?
(https://www.rabbitmq.com/man/rabbitmqctl.1.man.html)